stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -856,3 +875,50 @@ for func in (:log, :sqrt)
         end
     end
 end
+
+"""
+    hermitianpart(A, uplo=:U) -> Hermitian
+
+Return the Hermitian part of the square matrix `A`, defined as `(A + A') / 2`, as a
+[`Hermitian`](@ref) matrix. For real matrices `A`, this is also known as the symmetric part
+of `A`. The optional argument `uplo` controls the corresponding argument of the
+[`Hermitian`](@ref) view. For real matrices, the latter is equivalent to a
+[`Symmetric`](@ref) view.
+
+See also [`hermitianpart!`](@ref) for the corresponding in-place operation.
+
+!!! compat "Julia 1.10"
+    This function requires Julia 1.10 or later.
+"""
+hermitianpart(A::AbstractMatrix, uplo::Symbol=:U) = Hermitian(_hermitianpart(A), uplo)
+hermitianpart(a::Number) = _hermitianpart(a)
+
+"""
+    hermitianpart!(A, uplo=:U) -> Hermitian
+
+Overwrite the square matrix `A` with its Hermitian part `(A + A') / 2`, and return
+[`Hermitian(A, uplo)`](@ref). For real matrices `A`, this is also known as the symmetric
+part of `A`.
+
+See also [`hermitianpart`](@ref).
+
+!!! compat "Julia 1.10"
+    This function requires Julia 1.10 or later.
+"""
+hermitianpart!(A::AbstractMatrix, uplo::Symbol=:U) = Hermitian(_hermitianpart!(A), uplo)
+
+_hermitianpart(A::AbstractMatrix) = _hermitianpart!(copy_similar(A, Base.promote_op(/, eltype(A), Int)))
+_hermitianpart(a::T) where {T<:Number} = convert(Base.promote_op(/, T, Int), real(a))
```

Comment:
```suggestion
_hermitianpart(a::Number) = real(a)
```
As far as I can tell there is no reason to do a conversion here.

Reply:
I suggested doing so to ensure that `typeof(hermitianpart(x)) == eltype(hermitianpart([x;;]))` for `x::Number`

Reply:
When this is used along the diagonal of the matrix, then that conversion happens anyway, since the target array has that specific `eltype`. In that case, the diagonal would be filled with complex numbers with vanishing real part, not with real numbers! Ananlogously, computing `(a+a') == (a+conj(a))/2` for complex `a` yields a complex number, in contrast to `real(a)`. OTOH, there is the "operator real part" interpretation, which would perfectly allow the pure `real(a)` implementation. I believe, however, that we should remove the `hermitianpart(a::Number)` method, because it is (a) currently undocumented, and (b) never called internally (we call `_hermitianpart` instead).

Reply:
I removed `hermitianpart(::Number)` and the type conversion altogether, because the goal is to provide a "matrix symmetrizing function". `hermitianpart(::Number)` does not interfere with that goal, so can be an independent PR if necessary. People may then also have a concrete use case and expectations regarding the return type. It would also require an explicit comment in the docstring on how `hermitianpart` is working on `Number`s.